Update to the ISIS3 installation instructions

twilson271828 opened the pull request:
[index.txt](https://github.com/USGS-Astrogeology/ISIS3/files/2429945/index.txt)


GitHub would not allow me to attach the index.html page, so I changed the extension to text.
Just change the extension to html when you download it.  Need help on the Unix Environment Setup section near the end (or maybe we should just leave  that out, I'm not sure).

Also I think it could be helpful in the developer section to have a link to the isis3_cmake channel or one of the other channels ISIS devs hang out on (or ideally the discussion forums once those get set up).

twilson271828 committed:
Revert "Updated Install Guide for ISIS 3."

This reverts commit 8f18dfccd83191ef5c272caa2a64ce938c9606f7.

twilson271828 committed:
Updated the ISIS3 installation guide.

jessemapel commented:
Do we want to mention that you can run ISIS using the Ubuntu sub-system on windows? Kaj's been using it for a little while and it seems to work great.

jessemapel commented:
We are still supporting older systems for installation and use, Should we mention them under the OS requirements?

jessemapel commented:
I'd suggest we remove the mention of Jenkins because it's not up and running yet. If it's ready by the next release then we can mention it then.

jessemapel commented:
The UNIX environment part is going to depend on how we work with the new directory structure.

jessemapel commented:
Lastly, this never actually mentions how to download ISIS3.6.0, we need a section about how to install via anaconda. Is that intended to be under UNIX Environment Setup?

thareUSGS commented:
Perhaps could be updated (since it is over a year old): http://planetarygis.blogspot.com/2017/07/isis3-on-windows-10-bash.html

more recent but similar Wiki post which worked under an Ubuntu sub-system (on Windows 10) too: https://github.com/USGS-Astrogeology/ISIS3/wiki/Installing-ISIS3-via-Anaconda

twilson271828 commented:
Perhaps Kaj's help could be solicited in describing how he uses ISIS3 on Windows?  I'll add him as a reviewer.  I was thinking that installation via Anaconda would be under the Unix Environment Setup, but I think the title of that section is somewhat dated and should be changed.

scsides commented:
Suggest we tell which OSs it is built on and then in a separate section which OSs we test on. Then put the usual "it may work on others, but it is not tested"

scsides commented:
Update the requirements to be:
2.5GB for binaries
10 to 510GB of data

scsides commented:
In:

DTM Requirements
The strength of ISIS 3 strength 

strength is repeated

ihumphrey commented:
@twilson271828 Ya i think it would be useful to talk to Kaj about how he's set up his Windows Subsystem for Linux (WSL). Also, we could say it is possible to install and run isis3.6.0 on Windows 10's WSL, and provide a link to Microsoft's instructions for [setting up WSL](https://docs.microsoft.com/en-us/windows/wsl/install-win10).

AstroKEW78 commented:
On WSL I use the Ubuntu flavor, installed via the windows 10 app store.  
I then use Trent's directions:
http://planetarygis.blogspot.com/2017/07/isis3-on-windows-10-bash.html
(i.e. first get WSL going, install a couple of libraries, then rsync the public Ubuntu binaries and data)
Note that when major windows updates occur (e.g. from build 1703 to 1709), WSL also improves. Newer versions of Ubuntu are also available through the windows app store. I'm currently running Ubuntu 16 (Xenial), which works fine, but I think 18 is now available.

jessemapel commented:
@twilson271828 Putting the instructions in there sounds good. Renaming it also sounds like a good idea. I think it should also be all the way at the top of the documentation. If it's a guide to installing ISIS3, the first thing you should do is download the binaries. Then, you can download all of the data.

AstroKEW78 commented:
Maybe for windows instructions we could add links for Xming and VcXsrv,
Either Xming or VcXsrv (https://sourceforge.net/projects/vcxsrv/) works for x-windowing, but I like VcXsrv better since it is basically the xorg git sources compiled with Visual C++ 2012, and is actively being developed.

twilson271828 committed:
Updates to the installation guide as requested from reviewers.

twilson271828 commented:
Just updated the Installation Guide with the advice and comments from everyone. I included a link to Trent's blog post under the Windows 10 section and credit his blog (with his permission).  I also
included links to the windows X servers that Kaj recommended.  The ISIS3 conda installation process 
still needs to be finished, and is waiting on Ian to sort out some issues we were having.  The index.txt file I attached to this PR at the top now points to the most recent compiled version.  If you downloaded this PR, you can just go into $ISISROOT/src/docsys/documents/InstallGuide and cp Makefile-temp to Makefile.  Then just type make and it will compile into html for you without going through the process of making all of the documentation.

jlaura commented on isis/src/docsys/documents/InstallGuide/InstallGuide.xml:
```diff
@@ -24,30 +24,89 @@ you must <em>not</em> upgrade the ISIS Data Files!!!
     </ul>
 
+
+    <p>ISIS3 is also build and tested nightly on the following operating systems.</p>
```
:s/build/built/g

jlaura commented on isis/src/docsys/documents/InstallGuide/InstallGuide.xml:
```diff
@@ -24,30 +24,89 @@ you must <em>not</em> upgrade the ISIS Data Files!!!
+      <li>Mac OS X 10.11.6</li>
+    </ul>
+<p>
+    ISIS3 maybe run on other operating systems then those listed above,
```
:s/maybe/may be/g

jlaura commented on isis/src/docsys/documents/InstallGuide/InstallGuide.xml:
```diff
@@ -24,30 +24,89 @@ you must <em>not</em> upgrade the ISIS Data Files!!!
-      <li>2 GB memory</li>
-      <li>10 GB to 180 GB disk space for ISIS installation</li>
+      <li>2 GB RAM</li>
+      <li>2.5 GB for ISIS3 binaries</li>
```
2.5GB of what?

jlaura commented on isis/src/docsys/documents/InstallGuide/InstallGuide.xml:
```diff
@@ -24,30 +24,89 @@ you must <em>not</em> upgrade the ISIS Data Files!!!
+      <li>2.5 GB for ISIS3 binaries</li>
+      <li>10 GB to 510 GB disk space for ISIS3 data</li>
       <li>10 GB to many TB disk space for processing images</li>
       <li>A quality graphics card</li>
```
Really? Do we use the graphics card or is this because of QT?

twilson271828 replied:
There is an OpenGL dependency in QT which uses OpenGL to render the widgets.

jlaura commented on isis/src/docsys/documents/InstallGuide/InstallGuide.xml:
```diff
@@ -24,30 +24,89 @@ you must <em>not</em> upgrade the ISIS Data Files!!!
+<p>Note: More processors, memory, disk storage, and an additional graphcis card
+  (to support the use of two monitors at one time) may be useful depending on
+  the complexity of your processing requirements.
 </p>
```
Can we just remove? This is true for computers in general. More is better.

twilson271828 replied:
Yes, it's not really needed and sounds dated.

jlaura commented on isis/src/docsys/documents/InstallGuide/InstallGuide.xml:
```diff
@@ -24,30 +24,89 @@ you must <em>not</em> upgrade the ISIS Data Files!!!
+  <p>
+  While the ISIS3 development team has not examined these
+  instructions for correctness or completeness, they appear to have
+  successfully worked for a number of ISIS3 users within the USGS who run Windows.
```
I think we are beyond 'they appear'. Why not just say, 'please test and open an issue' or something to that extent. We are all one team in the building.

celbal replied:
Are we going to be supporting Windows? Is that a part of the ISIS support/maintenance deliverables this year?

Because if we aren't going to support it and we tell them to open an issue they are going to get ignored.

jlaura commented on isis/src/docsys/documents/InstallGuide/InstallGuide.xml:
```diff
@@ -24,30 +24,89 @@ you must <em>not</em> upgrade the ISIS Data Files!!!
+  instructions for correctness or completeness, they appear to have
+  successfully worked for a number of ISIS3 users within the USGS who run Windows.
+  Thanks for these instructions are directed towards Trent Hare who maintains an
+  excellent and thoughtful blog on <a href="http://planetarygis.blogspot.com/">Planetary GIS Science.</a>
```
not just excellent, but thoughtful :tada: :taco: for @thareUSGS

jlaura commented on isis/src/docsys/documents/InstallGuide/InstallGuide.xml:
```diff
@@ -24,30 +24,89 @@ you must <em>not</em> upgrade the ISIS Data Files!!!
+<A NAME="RunningOnWindowsWithSSH"> </A>
+<h3>Setting Up X11 forwarding on Windows with Putty and XMing</h3>
+<p>
+Some ISIS3 users prefer to run ISIS3 in Windows, but SSH into a computer
```
This is redundant. I wonder, do we usually tell our users how to use `ssh` and X forwarding. This seems like a great topic for a tutorial if someone wanted to put it in there. 

This comment is in part due to the SAT discussion of who the users we are targeting are. I think we can assume a basic level of linux/osx/win experience.

jlaura commented on isis/src/docsys/documents/InstallGuide/InstallGuide.xml:
```diff
@@ -24,30 +24,89 @@ you must <em>not</em> upgrade the ISIS Data Files!!!
+
    <h3>Mission Requirements</h3>
-   <p>ISIS 3 supports many planetary missions; in fact, over 40 different instruments including some
+   <p>ISIS3 supports many planetary missions; in fact, over 40 different instruments including some
```
Isn't this document about installing?

jlaura commented on isis/src/docsys/documents/InstallGuide/InstallGuide.xml:
```diff
@@ -59,43 +118,42 @@ one time) may be useful depending on the complexity of your processing requireme
 
    <h3>DTM Requirements</h3>
-   <p>The strength of ISIS 3 strength lies in its capabilities for planetary cartography.
+   <p>The strength of ISIS3 lies in its capabilities for planetary cartography.
```
As above, installing?

jlaura commented on isis/src/docsys/documents/InstallGuide/InstallGuide.xml:
```diff
@@ -59,43 +118,42 @@ one time) may be useful depending on the complexity of your processing requireme
 
-<p>To build and compile ISIS 3 requires following the instructions listed below, which are given on
-   the GitHub wiki page for the ISIS 3 project:
+<p>To build and compile ISIS3 requires following the instructions listed below, which are given on
```
:+1: Back to installing or building :+1:

jlaura commented on isis/src/docsys/documents/InstallGuide/InstallGuide.xml:
```diff
@@ -59,43 +118,42 @@ one time) may be useful depending on the complexity of your processing requireme
-    <li><a href="https://github.com/USGS-Astrogeology/ISIS3/wiki/Developer-Checklists">ISIS 3 Developer Checklist</a></li>
-    <li><a href="https://github.com/USGS-Astrogeology/ISIS3/wiki/Jenkins-CI">Jenkins Continuous Integration Testing</a></li>
-    <li><a href="https://github.com/USGS-Astrogeology/ISIS3/wiki/Managing-Custom-ISIS-Dependencies-(WIP)">Managing Custom ISIS 3 Dependencies</a></li>
+    <li><a href="https://github.com/USGS-Astrogeology/ISIS3/wiki/Developer-Checklists">ISIS3 Developer Checklist</a></li>
```
Is this installing?

jlaura commented on isis/src/docsys/documents/InstallGuide/InstallGuide.xml:
```diff
@@ -59,43 +118,42 @@ one time) may be useful depending on the complexity of your processing requireme
 <p>
     If you have the disk space and network speed, you may want to download all of the ISIS data areas which
     includes all missions supported by ISIS.   This takes over 130 GB of disk space!  If you only have a 10 Mbps
     network connection it will take nearly two days to download.  If you want to acquire only certain mission data
```
Does anyone have a 10Mbps connection these days? Maybe we can remove some of the narrative and just provide the facts? Then we could have a tutorial (if needed?) that is more narrative in style.

twilson271828 replied:
Deleted that part.  Also deleted the exclamation point.

jlaura commented on isis/src/docsys/documents/InstallGuide/InstallGuide.xml:
```diff
@@ -59,43 +118,42 @@ one time) may be useful depending on the complexity of your processing requireme
     includes all missions supported by ISIS.   This takes over 130 GB of disk space!  If you only have a 10 Mbps
     network connection it will take nearly two days to download.  If you want to acquire only certain mission data
-     <a href="#MissionSpecific">click here</a>.   To download all ISIS 3 data files continue reading.
+     <a href="#MissionSpecific">click here</a>.   To download all ISIS3 data files continue reading.
```
The above links are git oriented and I assume conda. This is a bit, `rsync` who now? Maybe we should alert users that data is shipped in a different way. So, I am being a total pain - complaining about narrative above and then talking about jarring here.

jlaura commented on isis/src/docsys/documents/InstallGuide/InstallGuide.xml:
```diff
@@ -382,22 +439,13 @@ one time) may be useful depending on the complexity of your processing requireme
     </p>
 
-<h3>How do I install ISIS 3.5.2 or earlier?</h3>
+<h3>How do I install ISIS3.5.2 or earlier?</h3>
```
:+1: Good to see this in here.

jlaura commented on isis/src/docsys/documents/InstallGuide/InstallGuide.xml:
```diff
@@ -424,10 +472,10 @@ one time) may be useful depending on the complexity of your processing requireme
-      binary version of ISIS 3.
+      binary version of ISIS3.
     </description>
     <author>Kristin Berry</author>
```
I think that it is awesome that @kberryUSGS took the lead and created (and/or modified) all of this. For the ISIS3 devs., does it make sense that this is by the ISIS3 development team?

Just today a whole bunch of folks (@twilson271828 @AstroKEW78 @thareUSGS @jessemapel - I am missing more I am sure) have contributed to this.

jessemapel replied:
@twilson271828 Has been taking the lead on updating this. This is an old document from before cmake. We should probably cite Tyler or the Astrogeology Science Center Software Developers.

jlaura replied:
Apologies @twilson271828. Thanks for taking the lead on this. If okay with you, how about ASC Software Developers? Just because of the team effort that you have been coordinating on this.

twilson271828 replied:
I'm okay with that.

twilson271828 committed:
More updates to the Installation Guide.

twilson271828 committed:
Updated history entry at bottom.

twilson271828 commented:
Do we need an isis3Startup.csh/isis3Startup.sh/isis3Startpy.py scripts in the scripts area (all of which do the same thing)?  Since we are distributing ISIS3 through conda, keeping the bash or python script makes sense to me but maintaining all three doesn't seem like a good idea.

twilson271828 commented:
@scsides 

`import os`
`os.readlink('/proc/%d/exe' % os.getppid() `

In my current environment, this gets the ppid of the parent process and outputs a string representation
'/usr/bin/tcsh' in my case, which can be parsed.  I'm not sure what it would do on a Mac though, or if this is portable to other systems.  
Anyway, that's one way to find out what your calling environment was when you called the script.  I think we can combine all of these into one python script.

jessemapel commented:
@twilson271828 :+1: for having a single Python script, because we're shipping a specific python environment by using Anaconda.

ihumphrey commented on isis/src/docsys/documents/InstallGuide/InstallGuide.xml:
```diff
@@ -382,22 +468,13 @@ one time) may be useful depending on the complexity of your processing requireme
-<h3>How do I install ISIS 3.5.2 or earlier?</h3>
+<h3>How do I install ISIS3.5.2 or earlier?</h3>
     <p>
          If you are looking for a version of ISIS3 prior to 3.5.3, please
```
This should say "prior to 3.6.0" instead of prior to "3.5.3"

twilson271828 replied:
Fixed it.

ihumphrey commented on isis/src/docsys/documents/InstallGuide/InstallGuide.xml:
```diff
@@ -24,30 +24,124 @@ you must <em>not</em> upgrade the ISIS Data Files!!!
+          source activate isis3alpha
+
+          #Download the ISIS3 version with label alpha
+          conda install -c usgs-astrogeology/label/alpha isis3
```
The "isis3alpha" will need to be changed to "isis3", and the install line would need to be:

```conda install -c usgs-astrogeology iss3```

This comment is here primarily for reference so we don't forget to update this section for the release candidate.

twilson271828 replied:
Fixed it.

ihumphrey commented on isis/src/docsys/documents/InstallGuide/InstallGuide.xml:
```diff
@@ -24,30 +24,124 @@ you must <em>not</em> upgrade the ISIS Data Files!!!
   <h2>Overview</h2>
 
+  <p>
+    This installation guide is for ISIS3 users interested in installing ISIS3 through conda, or compiling on their own systems using cmake.  
```
Should we mention the "compiling on their own systems using cmake"? That seems more developer-oriented.

It also may be useful to say that this installation guide is for ISIS3 users interesting in installing ISIS3 (3.6.0+) through conda.

twilson271828 replied:
I think we should delete that part.

SgStapleton commented on isis/src/docsys/documents/InstallGuide/InstallGuide.xml:
```diff
@@ -24,30 +24,124 @@ you must <em>not</em> upgrade the ISIS Data Files!!!
+      <li>Open a terminal window in the directory where you downloaded the script, and run the following commands (in this example, we chose to do a full install of Anaconda):</li>
+        <pre>
+          chmod +x Anaconda3-5.2.0-Linux-x86_64.sh
+          ./Anaconda3-5.2.0-Linux-x86_64.sh          
```
This command is OS-specific. Might want to mention that this is an example for the Linux platform.

twilson271828 replied:
Good catch!

twilson271828 committed:
Updates and edits to InstallGuide.xml, as well as the isis3VarInit script which initializes /scratch/isis3dev/isis//usgs/cpkgs/isis3/data//usgs/cpkgs/isis3/testData in the users conda environment.

twilson271828 commented:
Updated the InstallGuide.xml with suggestions from Ian and Summer.  I also added the python script isis3VarInit in $ISISROOT/scripts.  This initializes $ISISROOT/$ISIS3DATA/$ISIS3TESTDATA to default values within the conda environment.  I wrote it thinking that we might like to make these command line values that the user passes in somehow.

twilson271828 committed:
Added a few minor corrections to the InstallGuide.xml

SgStapleton commented on isis/src/docsys/documents/InstallGuide/InstallGuide.xml:
```diff
@@ -24,30 +24,132 @@ you must <em>not</em> upgrade the ISIS Data Files!!!
+          #Activate the environment
+          source activate isis3
+
+          #Download the ISIS3 version with label alpha
```
Is this still label alpha...?

SgStapleton commented on isis/src/docsys/documents/InstallGuide/InstallGuide.xml:
```diff
@@ -24,30 +24,132 @@ you must <em>not</em> upgrade the ISIS Data Files!!!
     <ul>
       <li>Ubuntu 18.04 LTS</li>
       <li>Mac OS X 10.13.6 High Sierra</li>
       <li>CentOS Linux 7.2.1511</li>
```
This is technically not true yet... Should we comment this out until we are sure?

SgStapleton commented on isis/src/docsys/documents/InstallGuide/InstallGuide.xml:
```diff
@@ -24,30 +24,132 @@ you must <em>not</em> upgrade the ISIS Data Files!!!
+
+    <ul>
+      <li>Ubuntu 14.04.3 LTS</li>
+      <li>Debian Linux 8 (jessie)</li>
```
It's never built on this system...

SgStapleton commented on isis/src/docsys/documents/InstallGuide/InstallGuide.xml:
```diff
@@ -24,30 +24,132 @@ you must <em>not</em> upgrade the ISIS Data Files!!!
+
+          #Run the source activate command a second time to process the changes:
+
+          source activate isis3alpha
```
Is this supposed to be simply ```source activate isis3```?

twilson271828 committed:
Added default parameter options to isis3VarInit, which can be overridden by the user.

twilson271828 commented:
I updated the isis3VarInit script, to allow the user to set $ISIS3DATA/$ISIS3TESTDATA.  The default option is to create these directories at the same level as $ISISROOT within the conda environment.

celbal commented on isis/scripts/isis3VarInit:
```diff
@@ -0,0 +1,48 @@
+parser = argparse.ArgumentParser(description='Usage:  ./isis3VarInit --data_dir <data dir path> --test_dir <test dir path')
+   
+isisroot = '$CONDA_PREFIX'
+data_dir='$CONDA_PREFIX/..data'
```
Does this need to be '$CONDA_PREFIX/../data'? Same with testData below

celbal commented on isis/src/docsys/documents/InstallGuide/InstallGuide.xml:
```diff
@@ -24,30 +24,132 @@ you must <em>not</em> upgrade the ISIS Data Files!!!
+        <pre>
+          #Add the following channels to the Anaconda installation
+          conda config --add channels conda-forge
+          conda config --add channels usgs-astrogeology
```
I believe this can be one command

twilson271828 replied:
conda config --add channels conda-forge --add-channels usgs-astrogeology

celbal commented on isis/src/docsys/documents/InstallGuide/InstallGuide.xml:
```diff
@@ -24,30 +24,132 @@ you must <em>not</em> upgrade the ISIS Data Files!!!
+          source activate isis3
+
+          #Download the ISIS3 version with label alpha
+          conda install -c usgs-astrogeology isis3
```
What alpha label? Does this need an alpha label or does the isis3 env from usgs-astrogeology have an alpha label?

twilson271828 replied:
That should have been removed.  Ian put a version of ISIS3 on conda with an alpha label for the purposes of testing these instructions.

celbal commented on isis/src/docsys/documents/InstallGuide/InstallGuide.xml:
```diff
@@ -24,30 +24,132 @@ you must <em>not</em> upgrade the ISIS Data Files!!!
+
+          #Run the source activate command a second time to process the changes:
+
+          source activate isis3alpha
```
When was this created or did isis3VarInit create it?

twilson271828 replied:
The isis3VarInit comand creates two bash shell scripts: 
 $CONDA_PREFIX/etc/conda/activate.d/env_vars.sh, and  $CONDA_PREFIX/etc/conda/deactivate.d/env_vars.sh which it sources when source activate <conda_env> is called.  You have to source activate a second time to create those variables ($ISISROOT, et. al) in memory.  It's kind of weird, I know.

celbal commented on isis/src/docsys/documents/InstallGuide/InstallGuide.xml:
```diff
@@ -24,30 +24,132 @@ you must <em>not</em> upgrade the ISIS Data Files!!!
+    <p>ISIS3 is also built and tested nightly on the following operating systems.</p>
+
+    <ul>
+      <li>Ubuntu 14.04.3 LTS</li>
```
Is this true any more?

twilson271828 replied:
Prog22 runs Ubuntu 14.04.3 and we're running tests on it nightly.

twilson271828 replied:
Never mind...just talked with Ian and we're not supporting Ubuntu 14.04 in the future so I'll delete the reference to it.

ihumphrey replied:
I would just remove this entire section ("ISIS3 is also built and tested nightly on the following operating systems"), since they are listed above (ubuntu18, macOS 10.13, CentOS 7, Fedora28)

celbal commented on isis/src/docsys/documents/InstallGuide/InstallGuide.xml:
```diff
@@ -59,64 +161,57 @@ one time) may be useful depending on the complexity of your processing requireme
 <p>
-    Remember to use the following command from the same directory you ran the previous rsync
-    command. In the example it was "/work1/isis3".
+    To download all ISIS3 data, enter the following commands at the command prompt:
```
Should there be a warning about how large the entire data set is?

twilson271828 replied:
There are people within Astro (like Kaj) who find these instructions useful.  Here I am emulating what the SPICE team does in their public disclaimer concerning wrappers like spiceypy - i.e. we're not supporting this, but it has been found useful for other users.  As long as we're clear that we don't support Windows 10 and will not be resolving bugs with this, then I think it's ok.

twilson271828 replied:
Yes, adding a line about how sizable the download is is a good idea.  It got lopped off in this draft in the name of conciseness.

celbal commented:
For the index.html file:
Line1: compiling on their own systems using cmake (shouldn't we just refer those people to the github repo? That is where the most current information is. We don't want to repeat info.)

twilson271828 committed:
Minor change to isis3VarInit script/updated documentation to InstallGuide.xml

twilson271828 committed:
Updated version of InstallGuide with Makayla's edits/suggestions.

ihumphrey commented on isis/src/docsys/documents/InstallGuide/InstallGuide.xml:
```diff
@@ -24,30 +24,139 @@ you must <em>not</em> upgrade the ISIS Data Files!!!
     <ul>
       <li>Ubuntu 18.04 LTS</li>
       <li>Mac OS X 10.13.6 High Sierra</li>
       <li>CentOS Linux 7.2.1511</li>
```
I would remove this, since as of now we don't have CentOS 7 support for development

twilson271828 committed:
Updated the install guide.

twilson271828 committed:
Added Janet's edits, updated isis3VarInit

twilson271828 committed:
Added py extension to isis3VarInit, updated InstallGuide.xml

SgStapleton reviewed (approved):
May need updates as we go, but looks like solid initial instructions.

ihumphrey commented:
![](https://github.trello.services/images/mini-trello-icon.png) [Installation documentation PR'd and ready for release candidate](https://trello.com/c/kz42NOWv/147-installation-documentation-prd-and-ready-for-release-candidate)
